authors/apps/authentication/social.py

```diff
@@ -1,33 +1,41 @@
+'''
+This module contains handlers for social authentication
+'''
 import facebook
-import os
 from google.auth.transport import requests
 from google.oauth2 import id_token
 from django.contrib.auth import authenticate
 from .models import User
 
 
 def register_social_user(email, username):
+    '''
+    This function handles user registration
+    and user login
+    '''
     user = User.objects.filter(email=email)
+    password = User.objects.make_random_password(length=20)
 
     if user.exists():
         current_user = User.objects.get(email=email)
+        current_user.set_password(password)
         current_user.is_active = True
         current_user.save()
```

Comment:
Great work @JamesMudidi. However, I wonder why we have to re-save the user if he/she already exists?